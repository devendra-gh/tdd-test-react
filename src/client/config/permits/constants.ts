import { IVariables } from '@tamm/app-composer';
import { PATH_ACCOUNT_UPGRADE, PATH_LOGIN } from './utils/constants/pageRoutes';

export const PROCESS_NAME = 'permits';

export const errorMsgMap: { [key: string]: string } = {
  'Incorrect emirates ID': 'errorMsg.incorrectemiratesid',
  'Incorrect Service Name': 'errorMsg.incorrectservicename',
  'Failed to send message': 'errorMsg.failedtosendmessage',
  'Incorrect DED license number': 'errorMsg.incorrectdedlicensenumber',
  'Incorrect pro mobile number': 'errorMsg.incorrectpromobilenumber',
  'Incorrect pro name': 'errorMsg.incorrectproname',
  'Incorrect email': 'errorMsg.incorrectemail',
  'Incorrect documents': 'errorMsg.incorrectdocuments',
  'Incorrect contact': 'errorMsg.incorrectcontact',
  'Incorrect Advertisement Details': 'errorMsg.incorrectadvertisementdetails',
  'Incorrect Advertisement Location': 'errorMsg.incorrectadvertisementlocation',
  'Incorrect Advertise Start Date': 'errorMsg.incorrectadvertisestartdate',
  'Incorrect Advertise End Date': 'errorMsg.incorrectadvertiseenddate',
  'Incorrect Customer Comment': 'errorMsg.incorrectcustomercomment',
  'Incorrect Study Specialization': 'errorMsg.incorrectstudyspecialization',
  'Incorrect Project Name': 'errorMsg.incorrectprojectname',
  'Incorrect Comments': 'errorMsg.incorrectcomments',
  'Incorrect Quantity': 'errorMsg.incorrectquantity',
  'Incorrect Plate Or Chassis Number': 'errorMsg.incorrectplateorchassisnumber',
  'Incorrect Seasonal Promotions': 'errorMsg.incorrectseasonalpromotions',
  'Incorrect Promotion Details': 'errorMsg.incorrectpromotiondetails',
  'Incorrect Promotion Location': 'errorMsg.incorrectpromotionlocation',
  'Incorrect Advertisements Quantity':
    'errorMsg.incorrectadvertisementsquantity',
  'Incorrect Plate Category': 'errorMsg.incorrectplatecategory',
  'Incorrect Advertisement Type': 'errorMsg.incorrectadvertisementtype',
  'Incorrect Note': 'errorMsg.incorrectnote',
  'Incorrect City': 'errorMsg.incorrectcity',
  'Incorrect Employee Comment': 'errorMsg.incorrectemployeecomment',
  'Incorrect Permit Type': 'errorMsg.incorrectpermittype',
  'Incorrect Airs': 'errorMsg.incorrectairs',
  'Incorrect Sign Boards': 'errorMsg.incorrectsignboards',
  'Please provide machines': 'errorMsg.pleaseprovidemachines',
  'Incorrect Umbrellas': 'errorMsg.incorrectumbrellas',
  'Incorrect Balloons': 'errorMsg.incorrectballoons',
  'Incorrect Sale Banners': 'errorMsg.incorrectsalebanners',
  'Incorrect Prizes': 'errorMsg.incorrectprizes',
  'Incorrect Propaganda Advertising Boards':
    'errorMsg.incorrectpropagandaadvertisingboards',
  'Incorrect Kiosks': 'errorMsg.incorrectkiosks',
  'Incorrect Vehicles': 'errorMsg.incorrectvehicles',
};

export const REQUIRES_SOP3 = {
  test: (props: IVariables) => {
    const { user } = props;
    if (user.Type === 'SOP3') return true;
    return false;
  },
  redirectTo: PATH_ACCOUNT_UPGRADE,
};

export const REQUIRES_CUSTOM_LOGIN = {
  test: (props: IVariables) => {
    if (props.loggedIn) return true;
    return false;
  },
  redirectTo: `${PATH_LOGIN}?redirectUrl=${window.location.href}`,
};
