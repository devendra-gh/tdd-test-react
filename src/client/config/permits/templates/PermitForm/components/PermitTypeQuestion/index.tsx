import React from 'react';
import { IVariables } from '@tamm/app-composer';
import FormTemplate from '@tamm/ui-lib-v2-form-template';
import PermitConfigs from 'client/config/permits/permitConfigs';

const permitCategories: { [key: string]: any[] } = {};
const permitConfigs: { [key: string]: any } = PermitConfigs;

Object.keys(PermitConfigs).forEach((permit: string) => {
  const {
    [permit]: { category, name },
  } = permitConfigs;

  let categories = permitCategories[category];
  if (!categories) {
    categories = [];
  }

  categories.push({
    id: name,
    label: name,
  });

  permitCategories[category] = categories;
});

export default (props: IVariables) => {
  // const { permitType, handleChange, selectedServiceType, i18n } = props;
  // if (!permitType) {
  //   return null;
  // }

  // const categoryPermits = permitCategories[permitType];

  return (
    <FormTemplate
      formData={{}}
      // i18n={i18n}
      id="permitType"
      inputGroups={[
        {
          fields: [
            // {
            //   'aria-label': 'field.serviceType.info',
            //   elementType: 'select',
            //   items: categoryPermits.map((sType: IVariables) => ({
            //     id: sType.id,
            //     label: i18n(`service.${sType.label}`),
            //   })),
            //   label: i18n('field.serviceType.info'),
            //   name: 'select',
            //   value: selectedServiceType,
            //   showSearch: false,
            //   onChange: handleChange,
            // },
          ],
          // name: i18n('title.selectServiceType'),
          // twoColumns: true,
        },
      ]}
    />
    //   <Select
    //   items={categoryPermits}
    //   value={selectedServiceType}
    //   label="Select Permit"
    //   placeholder="Select Permit Type"
    //   onChange={handleChange}
    // />
  );
};
