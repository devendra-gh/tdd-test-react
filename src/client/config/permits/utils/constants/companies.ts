export const companies: { [key: string]: Record<string, any> } = {
  DED: {
    label: 'company.label.ded',
    id: 'DED',
    type: '104',
  },
  FZ: {
    label: 'company.label.fz',
    id: 'FZ',
    type: '104',
  },
  NL: {
    label: 'company.label.nl',
    id: 'NL',
    type: '100',
  },
  ADGE: {
    label: 'company.label.adge',
    id: 'ADGE',
    type: '13',
  },
  FC: {
    label: 'company.label.fc',
    id: 'FC',
    type: '12',
  },
};
