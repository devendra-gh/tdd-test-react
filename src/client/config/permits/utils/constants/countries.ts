export const countries = [
  {
    name: 'country.unitedarabemirates',
    code: 971,
  },
];
