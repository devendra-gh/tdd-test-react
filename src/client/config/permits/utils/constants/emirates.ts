export const emirates = [
  {
    id: '1',
    label: 'emirate.abuDhabi',
  },
  {
    id: '2',
    label: 'emirate.dubai',
  },
  {
    id: '3',
    label: 'emirate.sharjah',
  },
  {
    id: '4',
    label: 'emirate.ajman',
  },
  {
    id: '5',
    label: 'emirate.ummAlQuwain',
  },
  {
    id: '6',
    label: 'emirate.rasAlKhaima',
  },
  {
    id: '7',
    label: 'emirate.fujairah',
  },
];
