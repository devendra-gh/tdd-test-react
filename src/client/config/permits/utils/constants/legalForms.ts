export const legalForms = [
  {
    id: 'Establishment',
    label: 'legalForm.establishment',
  },
  {
    id: 'General Partnership',
    label: 'legalForm.generalPartnership',
  },
  {
    id: 'Limited Liability Company',
    label: 'legalForm.limitedLiabilityCompany',
  },
  {
    id: 'Private Joint - Stock',
    label: 'legalForm.privateJointStock',
  },
  {
    id: 'Public Joint - Stock',
    label: 'legalForm.publicJointStock',
  },
  {
    id: 'Simple Limited Partnership',
    label: 'legalForm.simpleLimitedPartnership',
  },
  {
    id: 'Sole Proprietorship L.L.C.',
    label: 'legalForm.soleProprietorshipLLC',
  },
  {
    id: 'Sole Proprietorship PJSC',
    label: 'legalForm.soleProprietorshipPJSC',
  },
  {
    id: 'Branch',
    label: 'legalForm.branch',
  },
];
