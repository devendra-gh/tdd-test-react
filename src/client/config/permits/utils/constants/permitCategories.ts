const PERMIT_CATEGORY_SIGNBOARD_AD = 'signboardAd';
const PERMIT_CATEGORY_ADDITIONAL_AD = 'additionalAdPermit';
const PERMIT_CATEGORY_MACHINE_AD = 'machineAd';
const PERMIT_CATEGORY_MOBILE_AD = 'mobileAd';
const PERMIT_CATEGORY_PROMOTIONAL_AD = 'promotionalAd';
export {
  PERMIT_CATEGORY_SIGNBOARD_AD,
  PERMIT_CATEGORY_ADDITIONAL_AD,
  PERMIT_CATEGORY_MACHINE_AD,
  PERMIT_CATEGORY_MOBILE_AD,
  PERMIT_CATEGORY_PROMOTIONAL_AD,
};
