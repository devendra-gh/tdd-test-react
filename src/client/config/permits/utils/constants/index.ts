import { countries } from './countries';
import { legalForms } from './legalForms';
import { companies } from './companies';
import { emirates } from './emirates';
import { nationalities } from './nationalities';
import { permitType } from './permitType';
import { representativeType } from './representativeType';

export {
  companies,
  countries,
  emirates,
  legalForms,
  nationalities,
  permitType,
  representativeType,
};
