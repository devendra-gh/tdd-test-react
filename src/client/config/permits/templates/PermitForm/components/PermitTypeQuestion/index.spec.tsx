import React from 'react';
import { shallow } from 'enzyme';
import { PERMIT_CATEGORY_ADDITIONAL_AD } from 'client/config/permits/utils/constants/permitCategories';
import PermitTypeQuestion from './index';

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('PermitType Question', () => {
  let props: any;

  beforeEach(() => {
    props = {
      permitType: PERMIT_CATEGORY_ADDITIONAL_AD,
      handleChange: jest.fn(),
      selectedServiceType: 'air',
      i18n: jest.fn(),
    };
  });

  it('should successfully render the permit type question form', () => {
    const component = shallow(<PermitTypeQuestion {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should successfully render the permit type question form without permitType', () => {
    props.permitType = '';
    const component = shallow(<PermitTypeQuestion {...props} />);
    expect(component).toMatchSnapshot();
  });
});
