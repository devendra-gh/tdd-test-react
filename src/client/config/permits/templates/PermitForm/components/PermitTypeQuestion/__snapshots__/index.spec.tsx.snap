// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PermitType Question should successfully render the permit type question form 1`] = `
<LocalizedComponent(FormTemplate)
  cancelButton={null}
  cancelLink={null}
  formData={Object {}}
  i18n={[Function]}
  id="permitType"
  inputGroups={
    Array [
      Object {
        "fields": Array [],
      },
    ]
  }
  name={null}
  submitButton={null}
/>
`;

exports[`PermitType Question should successfully render the permit type question form without permitType 1`] = `
<LocalizedComponent(FormTemplate)
  cancelButton={null}
  cancelLink={null}
  formData={Object {}}
  i18n={[Function]}
  id="permitType"
  inputGroups={
    Array [
      Object {
        "fields": Array [],
      },
    ]
  }
  name={null}
  submitButton={null}
/>
`;
